rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/ChargingRSpace.scala

```diff
@@ -104,7 +104,7 @@ object ChargingRSpace {
             for {
               _ <- if (refundValue == Cost(0))
                     Sync[F].unit
-                  else cost.modify(_ + refundValue)
+                  else charge[F](Cost(-refundValue.value, "storage refund"))
```

Comment:
It looks like we are close to only ever interacting with `_cost` via the `charge` methods :+1: This will allow us to switch to a weaker typeclass soon :tada: